packages/loader/web-code-loader/src/utils.ts

```diff
@@ -53,3 +54,28 @@ export function extractPackageIdentifierDetails(codeDetailsPackage: string | IFl
         version,
     };
 }
+
+export function resolveFluidPackageEnvironment(
+    environment: IFluidPackageEnvironment,
+    baseUrl: string,
```

Comment:
Where does baseUrl come from?

Reply:
It's up to the code loader/resolver to get build this; this is a helper for resolvers. In our SemVerCdnCodeResolver it looks at the code details config for the cdn url, and then appends packageName@packageVersion to it